src/coreclr/jit/optimizer.cpp

```diff
@@ -4259,10 +4277,8 @@ void Compiler::optInvertWhileLoop(BasicBlock* block)
         }
     }
 
-    unsigned maxDupCostSz = 32;
+    unsigned maxDupCostSz = 34;
```

Comment:
> Have you looked at code size growth as a function of this number?

I haven't done a study, but it shouldn't be too hard to do. Although graphing against code size doesn't tell much about generated code performance, possibly.

> may be hard to assess with SPMI, given it will only show deltas for diffs

Wouldn't that be sufficient? If we want to know overall code size growth percentage, it wouldn't be hard to compute the total baseline code size once and apply the diff on top of that.

Might also be interesting to graph the number against things like CSEs, hoisted code, cloned loops.

Not sure how much time it's worth spending on something like this, though.

Reply:
>  graphing against code size doesn't tell much about generated code performance...

This is one of those "optimizations" that is really an enabling transformation -- not particularly beneficial on its own, possibly even a pessimization (at least in size, maybe not in cycles), but one that can lead to very profitable optimizations later.

So any sort of "growth vs threshold" chart would be interesting to see. If it's just the absolute growth and not the relative growth, fine... as you say we can figure out a rough baseline for the unchanged methods.

So my thought is that we should do this transformation as aggressively as we can without causing undue code size increases. I am curious where we sit currently on the spectrum (I don't really know) and whether there is some sort of inflection point or points out there possibly reflecting common language patters (say a dramatic jump in the curve at some threshold value)...